distributed/protocol/pickle.py

```diff
@@ -19,6 +23,59 @@
 pickle_types = tuple(pickle_types)
 
 
+@gen.coroutine
+def _tornado_coroutine_sample():
+    yield
+
+def is_tornado_coroutine(func):
+    """
+    Return whether *func* is a Tornado coroutine function.
+    Running coroutines are not supported.
+    """
+    return func.__code__ is _tornado_coroutine_sample.__code__
```

Comment:
How reliable is this check? What sort of change in `@gen.coroutine` would break this `is` check?